packages/backend/src/registries/CancellationTokenRegistry.ts

```diff
@@ -0,0 +1,59 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import type { CancellationTokenSource as ICancellationTokenSource, Disposable } from '@podman-desktop/api';
+import { CancellationTokenSource } from '../utils/CancellationToken';
+
+export class CancellationTokenRegistry implements Disposable {
+  #callbackId: number;
+  #callbacksCancellableToken: Map<number, ICancellationTokenSource>;
+
+  constructor() {
+    this.#callbackId = 0;
+    this.#callbacksCancellableToken = new Map<number, ICancellationTokenSource>();
+  }
+
+  createCancellationTokenSource(): number {
```

Comment:
@lstocchi

Reply:
This are the same methods use in PodmanDesktop, I wanted to stay close to their implementation

Reply:
I express myself badly. I was referring to the studioImpl. You only exposed the cancelToken but not the create. I would expect a `requestNewToken  `https://github.com/containers/podman-desktop-extension-ai-lab/pull/725/files#diff-3f3c50af66ddba22163580bcd3807fbb24558686b62bebd09e120bc6be130a76R394-R398

Reply:
IMO the frontend should not be responsible of creating those token, but should be attached to a task. 

Let's take the download model example, the `ModelManager` will create a cancellation token, set to the download task a `cancellationId` that the frontend will be able to see. But there is no reason why the front should create those tokens

Reply:
I'm not so convinced but we can always add it later 👍